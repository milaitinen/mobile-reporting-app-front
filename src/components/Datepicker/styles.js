import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
    $gray1: '$placeholder',
    $gray2: '$disabledPlaceholder',
    $disabledBg: '$disabled',
    $inputRadius: '$inputBorderRadius',
    $gray3: '$disabledBorder',
    $robotoLight: '$primaryFont',

    container: {
        width: 160,
        marginBottom: 10,
    },

    dateIcon: {
        color: '$inactive',
        paddingLeft: 5,
        paddingRight: 10,
        justifyContent: 'flex-end',
    },

    dateText: {
        color: '$placeholder',
        fontFamily: '$primaryFont',
    },

    disabledIcon: {
        color: '$disabledPlaceholder',
        paddingLeft: 5,
        paddingRight: 10,
        justifyContent: 'flex-end',
    }
});

export default styles;