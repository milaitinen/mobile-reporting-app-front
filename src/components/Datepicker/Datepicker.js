import DatePicker from 'react-native-datepicker';
import React, { Component } from 'react';
import styles from './styles';
import { Icon } from 'react-native-elements';


class Datepicker extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const { editable, mode, answer, onChange } = this.props;

        const text          = editable ? styles.$gray1 : styles.$gray2;
        const border        = editable ? 0 : 2;
        const background    = editable ? 'white' : styles.$disabledBg;
        const icon          = editable ? styles.dateIcon : styles.disabledIcon;

        if (mode === 'date') {
            return (
                <DatePicker
                    disabled={!editable}
                    style={styles.container}
                    customStyles={{
                        dateInput: {
                            borderWidth: border,
                            borderRadius: styles.$inputRadius,
                            backgroundColor: background,
                            borderColor: styles.$gray3
                        },
                        dateText: {
                            color: text,
                            fontFamily: styles.$robotoLight,
                        }
                    }}
                    date={answer}
                    mode={mode}
                    format="YYYY-MM-DD"
                    confirmBtnText="Confirm"
                    cancelBtnText="Cancel"
                    iconComponent={<Icon name={'calendar'} type={'evilicon'} size={40} iconStyle={icon}/>}
                    onDateChange={onChange}
                />
            );
        } else {
            return (
                <DatePicker
                    disabled={ !editable }
                    style={styles.container}
                    customStyles={{
                        dateInput: {
                            borderWidth: border,
                            borderRadius: styles.$inputRadius,
                            backgroundColor: background,
                            borderColor: styles.$gray3
                        },
                        dateText: {
                            color: text,
                            fontFamily: styles.$robotoLight,
                        }
                    }}
                    date={answer}
                    mode={mode}
                    format="HH:mm"
                    confirmBtnText="Confirm"
                    cancelBtnText="Cancel"
                    minuteInterval={10}
                    iconComponent={<Icon name={'clock'} type={'evilicon'} size={40} iconStyle={icon}/>}
                    onDateChange={onChange}
                />
            );
        }
    }
}

export default Datepicker;