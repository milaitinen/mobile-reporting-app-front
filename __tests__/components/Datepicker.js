import 'react-native';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Datepicker from '../../src/components/Datepicker/Datepicker';

configure({ adapter: new Adapter() });
jest.mock('DatePickerIOS', () => 'DatePickerIOS');
Date.now = jest.fn(() => new Date(Date.UTC(2018, 4, 12)).valueOf());

describe('Datepicker component', () => {
    it('renders correctly on date mode', () => {
        const tree = renderer.create(
            <Datepicker editable={true} mode={'date'} answer={'2018-04-22'} onChange={jest.fn()} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly on time mode', () => {
        const tree = renderer.create(
            <Datepicker editable={true} mode={'time'} answer={'2018-04-23'} onChange={jest.fn()} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when not editable', () => {
        const tree = renderer.create(
            <Datepicker editable={false} mode={'date'} answer={'2018-04-22'} onChange={jest.fn()} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

