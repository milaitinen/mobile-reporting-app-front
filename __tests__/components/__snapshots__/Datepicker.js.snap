// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker component renders correctly on date mode 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "width": 142,
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#aaa",
            "borderWidth": 1,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "white",
            "borderColor": undefined,
            "borderRadius": undefined,
            "borderWidth": 0,
          },
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "color": undefined,
              "fontFamily": undefined,
            },
          ]
        }
      >
        2018-04-22
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 40,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "EvilIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                pointerEvents="auto"
              >
                <DatePickerIOS
                  date={2018-04-21T21:00:00.000Z}
                  maximumDate={undefined}
                  minimumDate={undefined}
                  minuteInterval={undefined}
                  mode="date"
                  onDateChange={[Function]}
                  style={
                    Array [
                      Object {
                        "borderTopColor": "#ccc",
                        "borderTopWidth": 1,
                        "marginTop": 42,
                      },
                      undefined,
                    ]
                  }
                  timeZoneOffsetInMinutes={undefined}
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#666",
                      },
                      undefined,
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;

exports[`Datepicker component renders correctly on time mode 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "width": 142,
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#aaa",
            "borderWidth": 1,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "white",
            "borderColor": undefined,
            "borderRadius": undefined,
            "borderWidth": 0,
          },
          false,
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "color": undefined,
              "fontFamily": undefined,
            },
          ]
        }
      >
        20:18
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 40,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "EvilIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                pointerEvents="auto"
              >
                <DatePickerIOS
                  date={2018-04-12T17:18:00.000Z}
                  maximumDate={undefined}
                  minimumDate={undefined}
                  minuteInterval={10}
                  mode="time"
                  onDateChange={[Function]}
                  style={
                    Array [
                      Object {
                        "borderTopColor": "#ccc",
                        "borderTopWidth": 1,
                        "marginTop": 42,
                      },
                      undefined,
                    ]
                  }
                  timeZoneOffsetInMinutes={undefined}
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#666",
                      },
                      undefined,
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;

exports[`Datepicker component renders correctly when not editable 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "width": 142,
        },
        undefined,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#aaa",
            "borderWidth": 1,
            "flex": 1,
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": undefined,
            "borderColor": undefined,
            "borderRadius": undefined,
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#eee",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333",
            },
            Object {
              "color": undefined,
              "fontFamily": undefined,
            },
          ]
        }
      >
        2018-04-22
      </Text>
    </View>
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 40,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "EvilIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                pointerEvents="auto"
              >
                <DatePickerIOS
                  date={2018-04-21T21:00:00.000Z}
                  maximumDate={undefined}
                  minimumDate={undefined}
                  minuteInterval={undefined}
                  mode="date"
                  onDateChange={[Function]}
                  style={
                    Array [
                      Object {
                        "borderTopColor": "#ccc",
                        "borderTopWidth": 1,
                        "marginTop": 42,
                      },
                      undefined,
                    ]
                  }
                  timeZoneOffsetInMinutes={undefined}
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#666",
                      },
                      undefined,
                    ]
                  }
                >
                  Cancel
                </Text>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "padding": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ],
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#46cf98",
                        "fontSize": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
